### PR TITLE
[full-ci] disable federation capabilities

### LINKED ISSUE #4788: Capabilities indicate (currently) non-existing federation support
## Describe the bug

Capabilities currently return `true` for
```
files_sharing.federation.incoming
files_sharing.federation.outgoing
```

even though the server does not support federated sharing. The endpoint `/ocs/v2.php/apps/files_sharing/api/v1/remote_shares` only returns a 404 status.

## Steps to reproduce

1. Log into ocis instance
2. Retrieve capabilities
3. Retrieve remote_shares for an item 

## Expected behavior
The capabilities should reflect the actual capability of the server.

## Actual behavior
The capabilities do not match the availability of the federated sharing endpoint.

### LINKED ISSUE #4844: [Spaces] Change Space names in /graph/v1.0/me/drives to be more useful for clients
"Personal" Space and "Shares" Space have very strange names in the API response from `/graph/v1.0/me/drives`

```
            "driveAlias": "personal/mstingl",
            "driveType": "personal",
            "id": "1284d238-aa92-42ce-bdc4-0b0000009157$6b6253d5-cea6-4533-bd59-a34d160ffd24",
            "lastModifiedDateTime": "2022-08-19T07:36:07.599009932Z",
            "name": "Michael Stingl",
```

```
            "driveAlias": "virtual/shares",
            "driveType": "virtual",
            "id": "a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668",
            "lastModifiedDateTime": "2022-08-19T07:45:22.617232822Z",
            "name": "Shares Jail",
```

In the desktop and mobile clients, it should look like this:

CleanShot 2022-10-18 at 17 51 19@2x

Currently, desktop and mobile clients have "Personal" and "Shares" hardcoded, but it would be much better if the correct, user-facing names, would come from the API.

/cc @TheOneRing @felix-schwarz @kulmann @micbar

### LINKED ISSUE #4847: [Sharing] Announce new, roles-based sharing with a new capability for better client support
oCIS sharing based on roles, is very different from the oC10 sharing. To let clients support the new, roles-based oCIS sharing, a new capability is needed. The hack from ownCloud we, to always send the new oCIS roles AND the old oC10 bitmask, isn't a nice approach (oC10 ignores the roles, oCIS ignores the bitmask)

@micbar is this something quick?

/cc @TheOneRing @felix-schwarz @kulmann

### PR DESCRIPTION
**Pull request by micbar**
## Description

Fixes #4788 and includes a reva update to fix #4847 

## Example

```json
{
    "ocs": {
        "meta": {
            "status": "ok",
            "statuscode": 100,
            "message": "OK"
        },
        "data": {
            "capabilities": {
                "core": {
                    "pollinterval": 60,
                    "webdav-root": "remote.php/webdav",
                    "status": {
                        "installed": true,
                        "maintenance": false,
                        "needsDbUpgrade": false,
                        "version": "10.11.0.0",
                        "versionstring": "10.11.0",
                        "edition": "Community",
                        "productname": "Infinite Scale",
                        "product": "Infinite Scale",
                        "productversion": "2.0.0-beta.8+dev"
                    },
                    "support-url-signing": true
                },
                "checksums": {
                    "supportedTypes": [
                        "sha1",
                        "md5",
                        "adler32"
                    ],
                    "preferredUploadType": "SHA1"
                },
                "files": {
                    "privateLinks": false,
                    "bigfilechunking": false,
                    "undelete": true,
                    "versioning": true,
                    "favorites": false,
                    "blacklisted_files": [],
                    "tus_support": {
                        "version": "1.0.0",
                        "resumable": "1.0.0",
                        "extension": "creation,creation-with-upload",
                        "max_chunk_size": 100000000,
                        "http_method_override": ""
                    },
                    "archivers": [
                        {
                            "enabled": true,
                            "version": "2.0.0",
                            "formats": [
                                "tar",
                                "zip"
                            ],
                            "archiver_url": "/archiver",
                            "max_num_files": "10000",
                            "max_size": "1073741824"
                        }
                    ],
                    "app_providers": [
                        {
                            "enabled": true,
                            "version": "1.1.0",
                            "apps_url": "/app/list",
                            "open_url": "/app/open",
                            "open_web_url": "/app/open-with-web",
                            "new_url": "/app/new"
                        }
                    ]
                },
                "dav": {
                    "chunking": "",
                    "trashbin": "1.0",
                    "reports": [
                        "search-files"
                    ],
                    "chunkingParallelUploadDisabled": false
                },
                "files_sharing": {
                    "api_enabled": true,
                    "resharing": true,
                    "group_sharing": true,
                    "sharing_roles": true,
                    "auto_accept_share": true,
                    "share_with_group_members_only": true,
                    "share_with_membership_groups_only": true,
                    "search_min_length": 3,
                    "default_permissions": 22,
                    "user_enumeration": {
                        "enabled": true,
                        "group_members_only": true
                    },
                    "federation": {
                        "outgoing": false,
                        "incoming": false
                    },
                    "public": {
                        "enabled": true,
                        "send_mail": true,
                        "social_share": true,
                        "upload": true,
                        "multiple": true,
                        "supports_upload_only": true,
                        "password": {
                            "enforced_for": {
                                "read_only": false,
                                "read_write": false,
                                "upload_only": false
                            },
                            "enforced": false
                        },
                        "expire_date": {
                            "enabled": false
                        },
                        "can_edit": true,
                        "alias": true
                    },
                    "user": {
                        "send_mail": true,
                        "profile_picture": false,
                        "settings": [
                            {
                                "enabled": true,
                                "version": "1.0.0"
                            }
                        ]
                    }
                },
                "spaces": {
                    "version": "0.0.1",
                    "enabled": true,
                    "projects": true,
                    "share_jail": true
                }
            },
            "version": {
                "major": 10,
                "minor": 11,
                "micro": 0,
                "string": "10.11.0",
                "edition": "Community",
                "product": "Infinite Scale",
                "productversion": "2.0.0-beta.8+dev"
            }
        }
    }
}
```

## Partial fix
Parially fixes #4844 

```json
{
            "driveAlias": "virtual/shares",
            "driveType": "virtual",
            "id": "a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668",
            "lastModifiedDateTime": "2022-10-07T14:35:51.950965+02:00",
            "name": "Shares",
            "quota": {
                "remaining": 0,
                "state": "exceeded",
                "total": 0,
                "used": 0
            },
            "root": {
                "eTag": "\"5fb4ad8630f8485eb4fe03cc6b2fffb5\"",
                "id": "a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668",
                "webDavUrl": "https://localhost:9200/dav/spaces/a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668"
            },
            "webUrl": "https://localhost:9200/f/a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668"
        },
```

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by micbar** [comment]
@michaelstingl @felix-schwarz @TheOneRing FYI, see the example response.

**Comment by micbar** [comment]
@michaelstingl It also includes the space name change from `Share Jail` to `Shares`

File: services/frontend/pkg/config/config.go [hunk 1]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	UploadMaxChunkSize             int    `yaml:"upload_max_chunk_size" env:"FRONTEND_UPLOAD_MAX_CHUNK_SIZE" desc:"Sets the max chunk sizes in bytes for uploads via the frontend."`
+	UploadHTTPMethodOverride       string `yaml:"upload_http_method_override" env:"FRONTEND_UPLOAD_HTTP_METHOD_OVERRIDE" desc:"Advise TUS to replace PATCH requests by POST requests."`
+	DefaultUploadProtocol          string `yaml:"default_upload_protocol" env:"FRONTEND_DEFAULT_UPLOAD_PROTOCOL" desc:"The default upload protocol to use in the frontend (e.g. tus)."`
+	EnableResharing                bool   `yaml:"enable_resharing" env:"FRONTEND_ENABLE_RESHARING" desc:"Enables the support for resharing in the frontend."`
+	EnableFederatedSharingIncoming bool   `yaml:"enable_federated_sharing_incoming" env:"FRONTEND_ENABLE_FEDERATED_SHARING_INCOMING" desc:"Enables the support for federated sharing (incoming) in the frontend."`
+	EnableFederatedSharingOutgoing bool   `yaml:"enable_federated_sharing_outgoing" env:"FRONTEND_ENABLE_FEDERATED_SHARING_OUTGOING" desc:"Enables the support for federated sharing (outgoing) in the frontend."`
```
Comment (rhafer) [inline services/frontend/pkg/config/config.go:36]: Do we really need to add these knobs? This creates the impression that we support federated sharing by just setting these to true. ICBW, but to my understanding this is not the case, is it?

Wouldn't it be better to just hardcode the settings to `false` for now?

**Review by rhafer — CHANGES_REQUESTED** [review]

File: services/frontend/pkg/config/config.go (hunk 1 above)
Comment (micbar) [reply services/frontend/pkg/config/config.go:36]: Honestly IdK. To my knowledge CERN is using federation. With our codebase.

Comment (rhafer) [reply services/frontend/pkg/config/config.go:36]: Then we need to at least clarify in the documentation tags that this is just about the capabilties and not enabling any backend related stuff. "Enables the support for federated sharing (outgoing) in the frontend." sounds really missleading IMO.

Comment (micbar) [reply services/frontend/pkg/config/config.go:36]: okay. @mmattel FYI 
I will give it a try 😓

Comment (micbar) [reply services/frontend/pkg/config/config.go]: now better?

**Review by rhafer — APPROVED** [review]

File: services/frontend/pkg/config/config.go [hunk 2]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
-	EnableShareJail          bool   `yaml:"enable_share_jail" env:"FRONTEND_ENABLE_SHARE_JAIL" desc:"Indicates to clients that the share jail is supposed to be used."`
-	UploadMaxChunkSize       int    `yaml:"upload_max_chunk_size" env:"FRONTEND_UPLOAD_MAX_CHUNK_SIZE" desc:"Sets the max chunk sizes in bytes for uploads via the frontend."`
-	UploadHTTPMethodOverride string `yaml:"upload_http_method_override" env:"FRONTEND_UPLOAD_HTTP_METHOD_OVERRIDE" desc:"Advise TUS to replace PATCH requests by POST requests."`
-	DefaultUploadProtocol    string `yaml:"default_upload_protocol" env:"FRONTEND_DEFAULT_UPLOAD_PROTOCOL" desc:"The default upload protocol to use in the frontend (e.g. tus)."`
-	EnableResharing          bool   `yaml:"enable_resharing" env:"FRONTEND_ENABLE_RESHARING" desc:"Enables the support for resharing in the frontend."`
-	SearchMinLength          int    `yaml:"search_min_length" env:"FRONTEND_SEARCH_MIN_LENGTH" desc:"Minimum number of characters to enter before starting a search for Sharees."`
+	EnableFavorites                bool   `yaml:"enable_favorites" env:"FRONTEND_ENABLE_FAVORITES" desc:"Enables the support for favorites in the frontend."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:28]:
```suggestion
	EnableFavorites                bool   `yaml:"enable_favorites" env:"FRONTEND_ENABLE_FAVORITES" desc:"Enables support for favorites in the web frontend."`
```

File: services/frontend/pkg/config/config.go [hunk 3]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	EnableProjectSpaces            bool   `yaml:"enable_project_spaces" env:"FRONTEND_ENABLE_PROJECT_SPACES" desc:"Indicates to clients that project spaces are supposed to be made available."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:29]: Rereading this a couple of times, I dont get the point what is meant with it.

File: services/frontend/pkg/config/config.go [hunk 4]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	EnableShareJail                bool   `yaml:"enable_share_jail" env:"FRONTEND_ENABLE_SHARE_JAIL" desc:"Indicates to clients that the share jail is supposed to be used."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:30]: Add a sentence what is meant with share jail - else nobody knows what the impact is.

File: services/frontend/pkg/config/config.go [hunk 5]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	UploadMaxChunkSize             int    `yaml:"upload_max_chunk_size" env:"FRONTEND_UPLOAD_MAX_CHUNK_SIZE" desc:"Sets the max chunk sizes in bytes for uploads via the frontend."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:31]:
```suggestion
	UploadMaxChunkSize             int    `yaml:"upload_max_chunk_size" env:"FRONTEND_UPLOAD_MAX_CHUNK_SIZE" desc:"Sets the max chunk size in bytes for uploads via the web frontend."`
```

File: services/frontend/pkg/config/config.go [hunk 6]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	UploadHTTPMethodOverride       string `yaml:"upload_http_method_override" env:"FRONTEND_UPLOAD_HTTP_METHOD_OVERRIDE" desc:"Advise TUS to replace PATCH requests by POST requests."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:32]: What is the impact, maybe a small sentence would clarify it.

File: services/frontend/pkg/config/config.go [hunk 7]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	DefaultUploadProtocol          string `yaml:"default_upload_protocol" env:"FRONTEND_DEFAULT_UPLOAD_PROTOCOL" desc:"The default upload protocol to use in the frontend (e.g. tus)."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:33]: Missing: which values can be be set.

Comment (mmattel) [inline services/frontend/pkg/config/config.go:33]:
```suggestion
	DefaultUploadProtocol          string `yaml:"default_upload_protocol" env:"FRONTEND_DEFAULT_UPLOAD_PROTOCOL" desc:"Set the default upload protocol to use in the web frontend (e.g. tus)."`
Missing: which values can be be set.
```

File: services/frontend/pkg/config/config.go [hunk 8]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	EnableResharing                bool   `yaml:"enable_resharing" env:"FRONTEND_ENABLE_RESHARING" desc:"Enables the support for resharing in the frontend."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:34]:
```suggestion
	EnableResharing                bool   `yaml:"enable_resharing" env:"FRONTEND_ENABLE_RESHARING" desc:"Enables the support for resharing in the web frontend."`
```

File: services/frontend/pkg/config/config.go [hunk 9]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	EnableFederatedSharingIncoming bool   `yaml:"enable_federated_sharing_incoming" env:"FRONTEND_ENABLE_FEDERATED_SHARING_INCOMING" desc:"Enables the support for federated sharing (incoming) in the clients. This does not change the backend behavior."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:35]:
```suggestion
	EnableFederatedSharingIncoming bool   `yaml:"enable_federated_sharing_incoming" env:"FRONTEND_ENABLE_FEDERATED_SHARING_INCOMING" desc:"Enables support for incoming federated sharing  in for clients. The backend behaviour is not changed."`
```

File: services/frontend/pkg/config/config.go [hunk 10]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	EnableFederatedSharingOutgoing bool   `yaml:"enable_federated_sharing_outgoing" env:"FRONTEND_ENABLE_FEDERATED_SHARING_OUTGOING" desc:"Enables the support for federated sharing (outgoing) in the clients. This does not change the backend behavior."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:36]:
```suggestion
	EnableFederatedSharingOutgoing bool   `yaml:"enable_federated_sharing_outgoing" env:"FRONTEND_ENABLE_FEDERATED_SHARING_OUTGOING" desc:"Enables support for outgoing federated sharing  for clients. The backend behaviour is not changed."`
```

File: services/frontend/pkg/config/config.go [hunk 11]
```diff
@@ -25,14 +25,16 @@ type Config struct {
...
+	SearchMinLength                int    `yaml:"search_min_length" env:"FRONTEND_SEARCH_MIN_LENGTH" desc:"Minimum number of characters to enter before starting a search for Sharees."`
```
Comment (mmattel) [inline services/frontend/pkg/config/config.go:37]: What happens if zero is entered?
Is there a minimum or maximum value?

**Comment by mmattel** [comment]
In general:
* when there are possible values that can be entered, add the list in the description
* as I cant see it, are there default values set in the default field?
* as I cant see it, is the type field populated?

**Comment by micbar** [comment]
@mmattel You are correcting strings which are not part of the PR 😄 
I only added two vars, the others are already there.

File: services/frontend/pkg/config/config.go (hunk 4 above)
Comment (micbar) [reply services/frontend/pkg/config/config.go]: not supported, doesn't need documentation.

File: services/frontend/pkg/config/config.go (hunk 5 above)
Comment (micbar) [reply services/frontend/pkg/config/config.go]: please let us do this outside this PR

File: services/frontend/pkg/config/config.go (hunk 6 above)
Comment (micbar) [reply services/frontend/pkg/config/config.go:32]: different scope -> please extra PR/ticket

File: services/frontend/pkg/config/config.go (hunk 11 above)
Comment (micbar) [reply services/frontend/pkg/config/config.go]: Please let us not do that in this PR

**Review by mmattel — APPROVED** [review]

**Comment by micbar** [comment]
force merging because CI was already green.

**Pull request merged**